src/global/trampoline.rs

```diff
@@ -1,66 +1,297 @@
+/// Manifest data of the original executable.
+/// This is used by trampoline to set the environment variables
+/// prepened the path and execute the original executable.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct ManifestMetadata {
+    /// Path to the original executable.
+    pub exe: PathBuf,
+    /// Root path of the original executable that should be prepended to the PATH.
+    pub path: PathBuf,
+    /// Environment variables to be set before executing the original executable.
+    pub env: HashMap<String, String>,
+}
+
+impl ManifestMetadata {
+    /// Create a new manifest metadata.
+    pub fn new(exe: PathBuf, path: PathBuf, env: Option<HashMap<String, String>>) -> Self {
+        ManifestMetadata {
+            exe,
+            path,
+            env: env.unwrap_or_default(),
+        }
+    }
+
+    /// Read existing manifest metadata from the root path.
+    pub async fn from_root_path(
+        root_path: PathBuf,
+        exposed_name: &ExposedName,
+    ) -> miette::Result<Self> {
+        let manifest_path = root_path.join(exposed_name.to_string() + ".json");
+        let manifest_str = tokio_fs::read_to_string(manifest_path)
+            .await
+            .into_diagnostic()?;
+        serde_json::from_str(&manifest_str).into_diagnostic()
+    }
+}
+
+/// Represent an exposed global binary installed by pixi.
+/// This can be either a trampoline or a old script.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum GlobalBin {
+    Trampoline(Trampoline),
+    Script(PathBuf),
+}
+
+impl GlobalBin {
+    /// Return the path to the original executable.
+    pub async fn executable(&self) -> miette::Result<PathBuf> {
+        Ok(match self {
+            GlobalBin::Trampoline(trampoline) => trampoline.original_exe(),
+            GlobalBin::Script(script) => extract_executable_from_script(script).await?,
+        })
+    }
+
+    /// Return exposed name
+    pub fn exposed_name(&self) -> ExposedName {
+        match self {
+            GlobalBin::Trampoline(trampoline) => trampoline.exposed_name.clone(),
+            GlobalBin::Script(script) => {
+                ExposedName::from_str(&executable_from_path(script)).unwrap()
+            }
+        }
+    }
+
+    /// Return the path to the exposed binary.
+    pub fn path(&self) -> PathBuf {
+        match self {
+            GlobalBin::Trampoline(trampoline) => trampoline.path(),
+            GlobalBin::Script(script) => script.clone(),
+        }
+    }
+
+    /// Return if the exposed global binary is trampoline.
+    pub fn is_trampoline(&self) -> bool {
+        matches!(self, GlobalBin::Trampoline(_))
+    }
+
+    /// Return the inner trampoline.
+    pub fn trampoline(&self) -> Option<&Trampoline> {
+        match self {
+            GlobalBin::Trampoline(trampoline) => Some(trampoline),
+            _ => None,
+        }
+    }
+
+    /// Remove exposed global binary.
+    /// In case if it's trampoline, it will also remove it's manifest.
```

Comment:
```suggestion
    /// In case it is a trampoline, it will also remove its manifest.
```

I see you called it manifest throughout the code.

Maybe we could name the folder and variables in the code "index"?